plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/condition/NamedResourceDoesntExist.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v 1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.condition;
+
+import static org.junit.Assert.assertNotNull;
+
+import org.jboss.reddeer.common.condition.AbstractWaitCondition;
+import org.jboss.tools.openshift.core.connection.Connection;
+
+import com.openshift.restclient.OpenShiftException;
+
+/**
+ * Wait condition to wait for existence of a (core, openshift client-lib)
+ * resource of a given kind and name in the given project.
+ * 
+ * @author adietish@redhat.com
+ *
+ */
+public class NamedResourceDoesntExist extends AbstractWaitCondition {
+
+	private final String kind;
+	private final String name;
+	private String project;
+	private Connection connection;
+	
+	public NamedResourceDoesntExist(String kind, String name, String project, Connection connection) {
+		assertNotNull(kind);
+		assertNotNull(name);
+		assertNotNull(connection);
+
```

Comment:
missing assertNotNull for a project? What in case of null project? I've checked DefaultClient but it uses namespace all the time.